### PR TITLE
kernel: use ExecStatus instead of UInt where possible

### PR DESCRIPTION
**Pull request by fingolfin**
- kernel: turn ExecStatus into a proper enum (PR #4729)
- kernel: use ExecStatus instead of UInt where possible

This contains #4729. So either merge this PR to get both; or if PR #4729 is merged first, I can rebase this one afterwards.

### CODE REVIEW

File: src/common.h [hunk 1]
```diff
@@ -244,6 +244,28 @@ typedef struct TypInputFile TypInputFile;
...
 
 
+/****************************************************************************
+**
+*T  ExecStatus . . . .  type of status values returned by read, eval and exec
+**                      subroutines, explaining why evaluation, or execution
+**                      has terminated.
+**
+**  Values are powers of two; this is used to test a given status for
+**  multiple possible values simultaneously.
```
Comment (ChrisJefferson) [inline src/common.h:254]: this statement isn't true?

Comment (ChrisJefferson) [reply src/common.h:254]: on reading, I believe the plan is to remove this 'power of two' and not do bit-oring tricks (which I agree are better to get rid of), so it's just this comment that needs removing?

Comment (fingolfin) [reply src/common.h:254]: those two lines are removed in the first patch and then accidentally re-introduced in the second. I'll get rid of them.

Comment (fingolfin) [reply src/common.h]: Fixed now, thanks for pointing this out!

**Review by ChrisJefferson — APPROVED** [review]

**Pull request merged**